src/foam/dao/MDAO.java

```diff
@@ -63,6 +63,7 @@ public FObject find_(X x, Object o) {
   }
 
   public Sink select_(X x, Sink sink, long skip, long limit, Comparator order, Predicate predicate) {
+    sink = decorateSink_(sink,skip,limit,order,predicate);
```

Comment:
You shouldn't do this here. Inside the plan you should add whatever Sink decorators are required because you couldn't handle them efficiently, but it's too early to tell at this point which ones you can handle efficiently.